src/utils/decodedSourcemap.ts

```diff
@@ -22,7 +22,27 @@ export function decodedSourcemap(map: Input): ExistingDecodedSourceMap | null {
 		};
 	}
 
-	const mappings = typeof map.mappings === 'string' ? decode(map.mappings) : map.mappings;
+	const decodedMap = {
+		...(map as ExistingRawSourceMap | ExistingDecodedSourceMap),
+		mappings: undefined as unknown as SourceMapMappings
+	};
 
-	return { ...(map as ExistingRawSourceMap | ExistingDecodedSourceMap), mappings };
+	const origMappings = map.mappings;
+	let memoizedMappings: SourceMapMappings | undefined;
+
+	Object.defineProperty(decodedMap, 'mappings', {
+		enumerable: true,
+		get() {
+			if (memoizedMappings) {
+				return memoizedMappings;
+			}
```

Comment:
Memoizing the getter didn't really have any impact on performance from what I could tell, but it seemed like it would be more correct since reading from the `.mappings` property in multiple places is a perfectly reasonable thing to do.

Reply:
Maybe. But I assume this is the result of source map folding: When we have a chain of soutcemaps from plugins and some souremaps do not define certain source positions, the equivalent positions in other maps become irrelevant, so this is probably only possible at a very late stage. But maybe this is also a wrong conclusion.
In a case, great work! It would be nice to have a short comment here that this change greatly improves source map decoding.